packages/xo-server/src/api/vm.mjs

```diff
@@ -872,7 +872,18 @@ export async function convertToTemplate({ vm }) {
   // Convert to a template requires pool admin permission.
   await this.checkPermissions([[vm.$pool, 'administrate']])
 
-  await this.getXapiObject(vm).set_is_a_template(true)
+  const xapi = this.getXapi(vm)
+  const xapiVm = xapi.getObjectByRef(vm._xapiRef)
+
+  // Attempts to eject all removable media
+  const ignoreNotRemovable = error => {
+    if (error.code !== 'VBD_NOT_REMOVABLE_MEDIA') {
+      console.warn('convertToTemplate(%s)', xapiVm.uuid, error)
+    }
+  }
```

Comment:
So in any case, we still want to convert to template even if we didn't manage to eject the CDs?

Reply:
I believe so.

Reply:
Wouldn't it be a "surprising" behavior that when you convert a VM to a template, the template may or may not have CDs, depending on if something went wrong?

Reply:
Do you think it's preferable to rethrow the error?

I don't have a strong opinion on this and I will follow your suggestion :slightly_smiling_face:

Reply:
Yes, it would make more sense to me to rethrow. But then the `console.warn` would be pointless, right? So, just let it throw ⛄